### ISSUE TITLE
Document Resolvable Names

### ISSUE THREAD

**Issue by pnovotnak** [GoogleCloudPlatform/kubernetes#9682]
Document all SkyDNS records, or provide *and document* a method for getting an index from the Kubernetes API &| UI.

**Comment by erictune** [comment]
Each time you create a kubernetes service, it creates A and SRV records for that service.
So, I can't document all the records themselves, but the process is documented in `cluster/addons/dns/README.md`, which I admit is not the normal place you'd expect to find docs.

To dump out all the records, I did this:
```
ssh $ONE_OF_MY_CLUSTER_NODES
# apt-get install dnsutils
```
I first tried `dig -tAXFR @10.0.0.10 cluster.local` but it said `; Transfer failed.`

So then I tried `dig @10.0.0.10 cluster.local SRV` and `dig @10.0.0.10 cluster.local A` and I got some useful results.  I think that will give you an exhaustive list of records.

**Comment by pnovotnak** [comment]
Perfect! After running `kubectl describe service/kube-dns` to get the endpoint for the DNS service and querying it like `dig @<endpoint ip> cluster.local SRV` I indeed see an exhaustive list of services. I had tried AXFR previously and gotten the same result.

So, this brings me to my real question, which I can think a little more on and create a separate ticket for if appropriate, or buzz off if it's not in the cards.

Will endpoints ever have DNS addresses? I could be a long way from the mark here, but... The "What things get DNS names?" Section is clear that only services get IPs, but doesn't feel 100% clear on exactly which virtual IPs get addresses.

I'd like to create resources that are accessible within my private network at my cloud provider (though outside my Kubernetes cluster), but not to the outside internet, that are resolvable via DNS within the cluster. Ironically it seems like endpoint IPs fit the bill (although they change and are not resolvable) whereas service IPs, which are stable and what, are not accessible from outside the cluster.
